main.py

```diff
@@ -41,99 +40,175 @@ def check(a, b):
+            game_state.Paused = not game_state.Paused
         if event.type == pygame.QUIT or pygame.key.get_pressed()[pygame.K_ESCAPE]:
             pygame.quit()
             import sys
             sys.exit(0)
         for key in range(pygame.K_1, pygame.K_8):
             if pygame.key.get_pressed()[key]:
                 board = preset.place(board, int(pygame.key.name(key)), a, b)
+                board = update_board(board)
+                draw_board(board)
+        if pygame.key.get_pressed()[pygame.K_f]:
+            game_state.GPSLimit = not game_state.GPSLimit
+            #draw_gps_slider(((maths.log(game_state.GPS, game_state.TopGPS) + 1) / -3) * (
+            #Widgets.EndOfSlider - Widgets.StartOfSlider) +
+            #                Widgets.EndOfSlider, game_state.GPSLimit)
+            bottom_gps_log=maths.log(game_state.BottomGPS,game_state.TopGPS)
+            draw_gps_slider(Widgets.EndOfSlider-((maths.log(game_state.GPS,game_state.TopGPS)-bottom_gps_log)*
+                                                 (Widgets.EndOfSlider-Widgets.StartOfSlider))/
+                            (1-bottom_gps_log),game_state.GPSLimit)
         if pygame.key.get_pressed()[pygame.K_RIGHT]:
-            one_turn = True
+            game_state.OneTurn = True
+        else:
+            game_state.OneTurn = False
         if pygame.key.get_pressed()[pygame.K_RETURN]:
-            board = clean_board()
+            board = config.Board()
+            board = update_board(board)
+            draw_board(board)
+            board.Generations = 0
         if pygame.mouse.get_pressed()[0]:
-            board[a][b].birth(config.Square)
+            if board.Size * board.Width + board.Edge / 2 < x < board.Size * board.Width + Widgets.ButtonSize + \
+                            board.Edge / 2:  # within the button+GPS slider area
+                if y < Widgets.StartOfSlider:
+                    y = Widgets.StartOfSlider
+                elif y > Widgets.EndOfSlider:
+                    y = Widgets.EndOfSlider
+                    game_state.GPSLimit = True
+                draw_gps_slider(y, game_state.GPSLimit)
+                bottom_gps_log = maths.log(game_state.BottomGPS, game_state.TopGPS)
+                game_state.GPS = game_state.TopGPS ** (((1 - bottom_gps_log) * (Widgets.EndOfSlider - y) /
+                                                        (Widgets.EndOfSlider - Widgets.StartOfSlider)) + bottom_gps_log)
+            elif 0 <= a < board.Width + board.Cushion and 0 <= b < board.Height + board.Cushion:
+                board.Cell[a][b].birth(config.Square)
+                board = update_board(board)
+                draw_board(board)
         if pygame.mouse.get_pressed()[2]:
-            board[a][b].kill()
-    return board, paused, one_turn
+            board.Cell[a][b].kill()
+            board = update_board(board)
+            draw_board(board)
+    return board, game_state
 
 
-def draw_board():
+def draw_board(board):
```

Comment:
everything board related should go in the board class

Reply:
done